src/NavItem.js

```diff
@@ -28,7 +28,7 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.onSelect) {
+    if (this.props.onSelect && this.props.onSelect.name !== 'bound handleCollapse') {
```

Comment:
but yes, i'm not sure why onSelect would be expected to preventDefault. it would make more sense to allow default behavior to happen and put it on the user to ensure that they use (or don't use) anything that would trigger a behavior even when using onSelect.